cli/admin/src/main/java/io/pravega/cli/admin/dataRecovery/DurableLogRecoveryCommand.java

```diff
@@ -0,0 +1,248 @@
+import java.util.Map;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Loads the storage instance, recovers all segments from there.
+ */
+public class DurableLogRecoveryCommand extends DataRecoveryCommand implements AutoCloseable {
+    private static final int CONTAINER_EPOCH = 1;
+    private static final Duration TIMEOUT = Duration.ofMillis(100 * 1000);
+
+    private static final DurableLogConfig NO_TRUNCATIONS_DURABLE_LOG_CONFIG = DurableLogConfig
+            .builder()
+            .with(DurableLogConfig.CHECKPOINT_MIN_COMMIT_COUNT, 10000)
+            .with(DurableLogConfig.CHECKPOINT_COMMIT_COUNT, 50000)
+            .with(DurableLogConfig.CHECKPOINT_TOTAL_COMMIT_LENGTH, 1024 * 1024 * 1024L)
+            .build();
+    private static final ReadIndexConfig DEFAULT_READ_INDEX_CONFIG = ReadIndexConfig.builder().build();
+
+    private static final AttributeIndexConfig DEFAULT_ATTRIBUTE_INDEX_CONFIG = AttributeIndexConfig.builder().build();
+
+    private static final ContainerConfig DEFAULT_CONFIG = ContainerConfig
+            .builder()
+            .with(ContainerConfig.SEGMENT_METADATA_EXPIRATION_SECONDS, 10 * 60)
+            .build();
+
+    private static final ContainerConfig CONTAINER_CONFIG = ContainerConfig
+            .builder()
+            .with(ContainerConfig.SEGMENT_METADATA_EXPIRATION_SECONDS, (int) DEFAULT_CONFIG.getSegmentMetadataExpiration().getSeconds())
+            .build();
+
+    private static final WriterConfig WRITER_CONFIG = WriterConfig.builder().build();
+
+    private final ScheduledExecutorService executorService = ExecutorServiceHelpers.newScheduledThreadPool(100, "recoveryProcessor");
+    private final int containerCount;
+    private final StorageFactory storageFactory;
+    private final CuratorFramework zkClient;
+    private final Storage storage;
+    private BookKeeperLogFactory dataLogFactory = null;
+
+    /**
+     * Creates an instance of DurableLogRecoveryCommand class.
+     *
+     * @param args The arguments for the command.
+     */
+    public DurableLogRecoveryCommand(CommandArgs args) {
+        super(args);
+        this.containerCount = getServiceConfig().getContainerCount();
+        this.storageFactory = createStorageFactory(executorService);
+        this.storage = this.storageFactory.createStorageAdapter();
+        this.zkClient = createZKClient();
+    }
+
+
+    @Override
+    public void execute() throws Exception {
+        outputInfo("Container Count = %d", this.containerCount);
+
+        // Start a zk client and create a bookKeeperLogFactory
+        val bkConfig = getCommandArgs().getState().getConfigBuilder()
+                .include(BookKeeperConfig.builder().with(BookKeeperConfig.ZK_ADDRESS, getServiceConfig().getZkURL()))
+                .build().getConfig(BookKeeperConfig::builder);
+
+        this.dataLogFactory = new BookKeeperLogFactory(bkConfig, this.zkClient, executorService);
```

Comment:
Why don't you create this in the constructor? That's where you create everything else and you have everything else you need.

You can leave the initialization here.